Branch-SDK/Branch-SDK/BNCContentDiscoveryManager.m

```diff
@@ -441,65 +433,76 @@ - (void)indexContentWithUrl:(NSString *)url spotlightIdentifier:(NSString *)spot
-    //create-by-name a selector fot the init method we want
-    SEL initItemSelector = NSSelectorFromString(@"initWithUniqueIdentifier:domainIdentifier:attributeSet:");
-    //call the selector on the searchableItem with appropriate arguments
-    searchableItem = ((id (*)(id, SEL, NSString *, NSString *, id))[searchableItem methodForSelector:initItemSelector])(searchableItem, initItemSelector, spotlightIdentifier, BRANCH_SPOTLIGHT_PREFIX, attributes);
-  
-    //create an assignment method to set the expiration date on the searchableItem
-    SEL expirationSelector = NSSelectorFromString(@"setExpirationDate:");
-    //now invoke it on the searchableItem, providing the expirationdate
-    ((void (*)(id, SEL, NSDate *))[searchableItem methodForSelector:expirationSelector])(searchableItem, expirationSelector, expirationDate);
-  
-
-    Class CSSearchableIndexClass = NSClassFromString(@"CSSearchableIndex");
-    SEL defaultSearchableIndexSelector = NSSelectorFromString(@"defaultSearchableIndex");
-    id defaultSearchableIndex = ((id (*)(id, SEL))[CSSearchableIndexClass methodForSelector:defaultSearchableIndexSelector])(CSSearchableIndexClass, defaultSearchableIndexSelector);
-    SEL indexSearchableItemsSelector = NSSelectorFromString(@"indexSearchableItems:completionHandler:");
-    void (^__nullable completionBlock)(NSError *indexError) = ^void(NSError *__nullable indexError) {
-        if (callback || spotlightCallback) {
-            if (indexError) {
-                if (callback) {
-                    callback([BNCPreferenceHelper preferenceHelper].userUrl, indexError);
-                }
-                else if (spotlightCallback) {
-                    spotlightCallback(nil, nil, indexError);
-                }
-            }
-            else {
-                if (callback) {
-                    callback(url, nil);
-                }
-                else if (spotlightCallback) {
-                    spotlightCallback(url, spotlightIdentifier, nil);
-                }
-            }
+    // Not handling error scenarios because they are already handled upstream by the caller
+    if (url) {
+        if (callback) {
+            callback(url, nil);
+        } else if (spotlightCallback) {
+            spotlightCallback(url, spotlightIdentifier, nil);
         }
-    };
-    ((void (*)(id, SEL, NSArray *, void (^ __nullable)(NSError * __nullable error)))[defaultSearchableIndex methodForSelector:indexSearchableItemsSelector])(defaultSearchableIndex, indexSearchableItemsSelector, @[searchableItem], completionBlock);
+    }
 #endif
 }
 
+#pragma mark Delegate Methods
+
+- (void)userActivityWillSave:(NSUserActivity *)userActivity {
+    [userActivity addUserInfoEntriesFromDictionary:self.userInfo];
+}
+
+#pragma mark Helper Methods
+
+- (UIViewController *)getActiveViewController {
+    UIViewController *rootViewController = [UIApplication sharedApplication].keyWindow.rootViewController;
```

Comment:
To comply with extensions, can you use: `Class UIApplicationClass = NSClassFromString(@"UIApplication");`